docs/content/stable/api/ysql/the-sql-language/statements/ddl_alter_table.md

```diff
@@ -222,68 +236,68 @@ It quietly succeeds. Now `\d children` shows that the foreign key constraint `ch
 
 Add the specified constraint to the table.
 
+##### Table rewrites
+
+Adding a `PRIMARY KEY` constraint results in a full table rewrite of the main table and all associated indexes, which can be a potentially expensive operation. For more details about [table rewrites, see this section](#alter-table-operations-that-involve-a-table-rewrite).
+
+The reason for the table rewrite is the clustered storage by primary key that YugabyteDB uses to store rows and indexes. Tables without a `PRIMARY KEY` have a hidden one underneath and rows are stored clustered on it. These rows need to be rewritten to use the newly added primary key column. 
```

Comment:
```suggestion
The table rewrite is needed because of how YugabyteDB stores rows and indexes. In YugabyteDB, data is distributed based on the primary key; when a table does not have an explicit primary key assigned, YugabyteDB automatically creates an internal row ID to use as the table's primary key. As a result, these rows need to be rewritten to use the newly added primary key column. For more information, refer to [Primary keys](../../../../../develop/data-modeling/primary-keys-ysql). 
```

Reply:
Thanks this is much more clear!